File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> myStack = new Stack<Character>();
        HashMap<Character, Character> myHashMap = new HashMap<Character, Character>();
        myHashMap.put('(', ')');
        myHashMap.put('{', '}');
        myHashMap.put('[', ']');
        HashSet<Character> myHashSet = new HashSet<Character>();
        myHashSet.add('(');
        myHashSet.add('[');
        myHashSet.add('{');
        
        for (int i = 0; i < s.length(); i++) {
            if (myHashSet.contains(s.charAt(i))) {
                myStack.push(s.charAt(i));
            } else {
                if (myStack.size() > 0) {
                    if (s.charAt(i) == myHashMap.get(myStack.get(myStack.size() - 1))) {
                        myStack.pop();
                    } else {
                        return false;
                    }
                } else {
                    return false;
                }
            }
        }

        return myStack.size() == 0 ? true : false;
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    let stack = [];
    let hashmap = new Map();
    hashmap.set("(", ")").set("{", "}").set("[", "]");
    let hashset = new Set();
    hashset.add("(").add("[").add("{");

    for (let i = 0; i < s.length; i++) {
        if (hashset.has(s[i])) {
            stack.push(s[i]);
        } else {
            if (hashmap.get(stack[stack.length - 1]) === s[i]) {
                stack.pop();
            } else {
                return false;
            }
        }
    }
    return stack.length === 0 ? true : false;
};